File: src/service/ProductService.js
import axios from "axios"

const API_URL = "http://localhost:5678/api/v1/product"

class ProductService{
     saveProduct(product){
        return axios.post(API_URL + "/create",product);
     }
     getAllProduct(){
        return axios.get(API_URL + "/read");
     }
     getAllProductById(id){
        return axios.get(API_URL + "/read/" + id);
     }
     deleteProduct(id){
        return axios.delete(API_URL +  "/delete/" + id);
     }
     updateProduct(product){
        return axios.put(API_URL + "/update/" +product.id, product);
     }
}

export default new ProductService

File: backend/src/main/java/com/product/ProductManagement/service/ProductService.java
package com.product.ProductManagement.service;

import java.util.List;

import com.product.ProductManagement.model.Product;

public interface ProductService {
    public Product saveProduct(Product product);
    public List<Product> getAllProducts();
    public Product getProductById(int id);
    public String deleteProduct(int id);
    public Product updateProduct(int id,Product product);

}
